lucien-ios/APIClient/LucienAPIClient.swift

```diff
@@ -93,4 +93,59 @@ final class LucienAPIClient: APIClient {
             }
         }
     }
+
+    func createPhotoURL(completion: @escaping (Result<AmazonS3ImageURL>) -> Void) {
+        let lucienRequest = LucienRequest.createPhotoURL
+        let urlRequest = lucienRequest.urlRequest
+        sendRequest(urlRequest) { response in
+            switch response {
+            case .success(let result):
+                guard let result = result else {
+                    return completion(.failure(LucienAPIError.noResult))
+                }
+                let decoder = JSONDecoder()
+                decoder.dateDecodingStrategy = .iso8601
+                guard let s3ImageURL = try? decoder.decode(AmazonS3ImageURL.self, from: result) else {
```

Comment:
In this case, rather than using the optional `try`, wrap the decode call in a `do-catch` and in the `catch` block, call the completion with the error returned instead of `cannotDecode`. Take a look at `getCurrentUser` for an example